### PR TITLE
support vector particular dispatches for sev

### LINKED ISSUE #1821: Faster hyperrectangle support vector for SingleEntryVector
Follow-up of https://github.com/JuliaReach/LazySets.jl/pull/1785

### PR DESCRIPTION
**Pull request by dfcaporale**
Closes #1821.

### CODE REVIEW

File: src/Interfaces/AbstractHyperrectangle.jl [hunk 1]
```diff
@@ -273,6 +273,20 @@ function σ(d::AbstractVector{N}, H::AbstractHyperrectangle{N}) where {N<:Real}
...
 end
 
+# helper function for single entry vector
+function _σ_sev_hyperrectangle(d::SingleEntryVector{N}, H::AbstractHyperrectangle{N}) where {N<:Real}
+
+    @assert d.n == dim(H) "a $(d.n)-dimensional vector is " *
+                                "incompatible with a $(dim(H))-dimensional set"
+    s = copy(center(H))
+    if d.v < zero(N)
+      s[d.i] -= radius_hyperrectangle(H, d.i)
```
Comment (mforets) [inline src/Interfaces/AbstractHyperrectangle.jl]: For indentation we use 4 spaces instead of 2.

**Comment by mforets** [comment]
Since only one value of the center is changed, instead of all of them, this PR changes the point which is returned, from a vertex in a corner, to the point aligned with the direction of the ball. I don't consider this breaking though, as we don't guarantee that the point is always the same for different implementations. For instance,

```julia
julia> σ(d, b) # here d is the single entry vector [1.0, 0.0]
2-element Array{Float64,1}:
 1.0
 0.0

julia> σ([1.0, 0.0], b) # same result as in master for d
2-element Array{Float64,1}:
 1.0
 1.0
```

**Comment by schillic** [comment]
> Since only one value of the center is changed, instead of all of them, this PR changes the point which is returned, from a vertex in a corner, to the point aligned with the direction of the ball. I don't consider this breaking though, as we don't guarantee that the point is always the same for different implementations.

Agreed. That is why a test should not rely on those values.

**Comment by mforets** [comment]
> Tests with SEVs would be nice. Example:

I added this test in the last commit.

**Pull request merged**